app/controllers/programs_controller.rb

```diff
@@ -6,4 +6,30 @@ def index
   def show
     @program = Program.find(params[:id])
   end
+
+  def create
+    @program = Program.new(program_params)
+
+    if @program.save
+      flash[:notice] = "Program saved"
+      redirect_to program_path(@program)
+    else
+      flash[:notice] = "Fill out form correctly"
+      render :new
+    end
+  end
+
+  def new
+    if current_admin.nil?
+      redirect_to root_path
```

Comment:
The user should get some indication that they aren't permitted to use a given feature, utilizing flash messages.